vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSlideController.java

```diff
@@ -79,21 +86,33 @@ public String slide(Model model, @PathVariable("slideId") String slideId, @PathV
         model.addAttribute("firstSlide", module.getStartSequence().getSlides().get(0).getId());
         String nextSlideId = "";
         String prevSlideId = "";
-        int slideIndex = 0;
-        slideIndex = sequenceSlides.indexOf(sildeManager.getSlide(slideId));
+        ISlide currentSlide = sildeManager.getSlide(slideId);
+        int slideIndex = sequenceSlides.indexOf(currentSlide);
         int slideSize = sequenceSlides.size();
         if (slideSize > slideIndex + 1) {
             nextSlideId = sequenceSlides.get(slideIndex + 1).getId();
         }
         if (slideIndex > 0) {
             prevSlideId = sequenceSlides.get(slideIndex - 1).getId();
         }
+        model.addAttribute("sequences",moduleManager.getModuleSequences(moduleId));
         model.addAttribute("sequence",sequenceExist);
         model.addAttribute("slides", sequenceSlides);
         model.addAttribute("currentSequenceId", sequenceId);
         model.addAttribute("nextSlide", nextSlideId);
         model.addAttribute("prevSlide", prevSlideId);
-        model.addAttribute("currentSlideCon", sildeManager.getSlide(slideId));
+        model.addAttribute("currentSlideCon", currentSlide);
+        if(currentSlide instanceof BranchingPoint) {
+            model.addAttribute("choices", ((BranchingPoint)currentSlide).getChoices());
+        }
+        if(back) {
```

Comment:
you only want to do this if the current slide is a branching point, otherwise it is not possible that the user clicked the back to branching point button.

Reply:
current slide can be a branching point if the user clicks on choice and the first slide of that choice is a branching point?

Reply:
what I mean is:
- this controller is called when the new slide page is rendered, right? 
- if back is true, then the user came from another sequence and clicked the 'back to bp' button
- the back to bp button will always bring the user back the the previous bp
- so if the current slide is not a bp, then there was an error somewhere, because that can't be.